hapi-fhir-jpaserver-mdm/src/main/java/ca/uhn/fhir/jpa/mdm/svc/MdmEidUpdateService.java

```
@@ -80,10 +80,15 @@ void handleMdmUpdate(IAnyResource theTargetResource, MatchedGoldenResourceCandid
 		} else {
 			//This is a new linking scenario. we have to break the existing link and link to the new Golden Resource. For now, we create duplicate.
 			//updated patient has an EID that matches to a new candidate. Link them, and set the Golden Resources possible duplicates
-			linkToNewGoldenResourceAndFlagAsDuplicate(theTargetResource, theMatchedGoldenResourceCandidate.getMatchResult(), updateContext.getExistingGoldenResource(), updateContext.getMatchedGoldenResource(), theMdmTransactionContext);
+			IAnyResource theOldGoldenResource = updateContext.getExistingGoldenResource();
+			if (theOldGoldenResource == null) {
+				ourLog.error("Old golden resource was null while updating MDM links with new golden resource. It is likely that a $mdm-clear was performed without a $mdm-submit. Link will not be updated.");
```

Why did you decide not to throw another Exception here?   The client will still see the NullPointerException in the logs as before, just with an extra error message before.   I see no reason to continue execution here.

---

The NPE is thrown in `linkToNewGoldenResourceAndFlagAsDuplicate()` in the else statement, so it will not be seen by the client. I can definitely change to an exception with HAPI error code though

---

If I add a HAPI error code, I think I will also change it to an exception, as it seems like the HAPI error codes are only used with exceptions and not with log error messages

---

I'm suggesting that you throw an InternalErrorException in this case, with a HAPI-****, rather than to simply log an error.